UnitTestFrameworkPkg/Library/Posix/MemoryAllocationLibPosix/MemoryAllocationLibPosix.c

```diff
@@ -293,13 +304,25 @@ FreeAlignedPages (
 
   @return  A pointer to the allocated buffer or NULL if allocation fails.
 
-**/VOID *
+**/
+VOID *
 EFIAPI
 AllocatePool (
   IN UINTN  AllocationSize
   )
 {
-  return malloc (AllocationSize);
+  POOL_HEAD  *PoolHead;
+
+  PoolHead = malloc (sizeof (POOL_HEAD) + AllocationSize);
+  if (PoolHead == NULL) {
+    return NULL;
+  }
+
+  DEBUG_CLEAR_MEMORY (PoolHead, AllocationSize);

```

Comment:
This fails to clear the last 8 bytes (size of POOL_HEAD).

Reply:
Thanks!  Really good catch.  I have pushed an update with this fixed.